include/ginkgo/core/base/executor.hpp

```diff
@@ -953,6 +992,130 @@ using DefaultExecutor = CudaExecutor;
 }  // namespace kernels
 
 
+/**
+ * This is the Executor subclass which represents the HIP enhanced device.
+ *
+ * @ingroup exec_hip
+ * @ingroup Executor
+ */
+class HipExecutor : public detail::ExecutorBase<HipExecutor>,
+                    public std::enable_shared_from_this<HipExecutor> {
+    friend class ExecutorBase<HipExecutor>;
+
+public:
+    /**
+     * Creates a new HipExecutor.
+     *
+     * @param device_id  the HIP device id of this device
+     * @param master  an executor on the host that is used to invoke the device
+     * kernels
```

Comment:
nit: Can you indent this line, so it is clear it belongs to the `@param master`?